### ISSUE TITLE
Prepare new embed system for public release

### ISSUE THREAD

**Issue by charrondev** [vanilla/vanilla#8299]
The new embed system (`EmbedManager`) needs to be prepared for public release.

There a few things that need to be done before it is ready however. Each of these should be broken into its own issue and linked to this epic.

- Create a cleaner interface for the `Embed` class. Some of the methods could use clearer names.
- Make a clearer distinction between what data can be parsed out of an input URL and what data requires fetching from another server. Eg. Some have networking as a requirement. I'm thinking:
  - The embed class itself is an entity object that uses Garden\Schema to enforce its data structure, and any that require a network request should have a factory for their creation.
- Cleanup the structure of embeds that get mounted on the frontend. There should be a clear way to structure their HTML so that they can be mounted cleanly by a react component on the frontend.
- Create new public documentation on the new embed system.

**Comment by charrondev** [comment]
Blocking https://github.com/vanillaforums/estimates/pull/547
